test/remoting.integration.js

```diff
@@ -238,5 +185,185 @@ describe('remoting - integration', function() {
+        'exists(id:any):boolean GET /stores/:id/exists',
+        'findById(id:any,filter:object):store GET /stores/:id',
+        'replaceById(id:any,data:object):store PUT /stores/:id',
+        'replaceById(id:any,data:object):store POST /stores/:id/replace',
+        'find(filter:object):store GET /stores',
+        'findOne(filter:object):store GET /stores/findOne',
+        'updateAll(where:object,data:object):object POST /stores/update',
+        'deleteById(id:any):object DELETE /stores/:id',
+        'count(where:object):number GET /stores/count',
+        'prototype.updateAttributes(data:object):store PATCH /stores/:id',
+        'createChangeStream(options:object):ReadableStream POST /stores/change-stream',
+      ];
+
+      // TODO: update the doc for list of methods accordingly
+      // https://docs.strongloop.com/display/public/LB/Exposing+models+over+REST
+      expect(methods).to.include.members(expectedMethods);
+    });
   });
 });
+
+describe('With model.settings.replaceOnPUT false' +
+  'set to false', function() {
+  lt.beforeEach.withApp(app);
+  lt.beforeEach.givenModel('storeWithReplaceOnPUTfalse');
+  afterEach(function(done) {
+    this.app.models.storeWithReplaceOnPUTfalse.destroyAll(done);
+  });
+
+  it('should have expected remote methods',
+  function() {
+    var storeClass = findClass('storeWithReplaceOnPUTfalse');
+    var methods = getMethods(storeClass.methods);
+
+    var expectedMethods = [
+      'create(data:object):storeWithReplaceOnPUTfalse POST /stores-updating',
+      'upsert(data:object):storeWithReplaceOnPUTfalse PATCH /stores-updating',
+      'upsert(data:object):storeWithReplaceOnPUTfalse PUT /stores-updating',
+      'replaceOrCreate(data:object):storeWithReplaceOnPUTfalse POST /stores-updating/replaceOrCreate',
+      'exists(id:any):boolean GET /stores-updating/:id/exists',
+      'findById(id:any,filter:object):storeWithReplaceOnPUTfalse GET /stores-updating/:id',
+      'replaceById(id:any,data:object):storeWithReplaceOnPUTfalse POST /stores-updating/:id/replace',
+      'find(filter:object):storeWithReplaceOnPUTfalse GET /stores-updating',
+      'findOne(filter:object):storeWithReplaceOnPUTfalse GET /stores-updating/findOne',
+      'updateAll(where:object,data:object):object POST /stores-updating/update',
+      'deleteById(id:any):object DELETE /stores-updating/:id',
+      'count(where:object):number GET /stores-updating/count',
+      'prototype.updateAttributes(data:object):storeWithReplaceOnPUTfalse PATCH /stores-updating/:id',
+      'prototype.updateAttributes(data:object):storeWithReplaceOnPUTfalse PUT /stores-updating/:id',
+      'createChangeStream(options:object):ReadableStream POST /stores-updating/change-stream',
+    ];
+
+    // The list of methods is from docs:
+    // https://docs.strongloop.com/display/public/LB/Exposing+models+over+REST
+    expect(methods).to.include.members(expectedMethods);
+  });
+});
+
+describe('With model.settings.replaceOnPUT true' +
```

Comment:
What is the difference between this test and `it('has expected remote methods with model.settings.replaceOnPUT set to true'` above? I think it's enough to keep only one of these tests, isn't it?